packages/visual-editor/src/utils/simulateDnD.ts

```diff
@@ -0,0 +1,59 @@
+import { CTFL_DRAGGING_ELEMENT, NEW_COMPONENT_ID } from '../types/constants';
+import DragState from './dragState';
+
+class SimulateDnD extends DragState {
+  private draggingElement: Element | null;
+
+  constructor() {
+    super();
+    this.draggingElement = null;
+  }
+
+  public setupDrag() {
+    this.updateIsDragStartedOnParent(true);
+  }
+
+  public startDrag(coordX: number, coordY: number) {
+    this.draggingElement = document.getElementById(NEW_COMPONENT_ID);
+    this.updateIsDragging(true);
+    this.simulateMouseEvent(coordX, coordY, 'mousedown');
+  }
+
+  public updateDrag(coordX: number, coordY: number) {
+    if (!this.draggingElement) {
+      this.draggingElement = document.querySelector(`[${CTFL_DRAGGING_ELEMENT}]`);
+    }
+    this.simulateMouseEvent(coordX, coordY);
+  }
+
+  public endDrag(coordX: number, coordY: number) {
+    this.simulateMouseEvent(coordX, coordY, 'mouseup');
+    this.reset();
+  }
+
+  public reset() {
+    this.draggingElement = null;
+    this.resetState();
+  }
+
+  private simulateMouseEvent(coordX: number, coordY: number, eventName = 'mousemove') {
+    const options = {
+      bubbles: true,
+      cancelable: true,
+      view: window,
+      pageX: 0,
+      pageY: 0,
+      clientX: coordX,
+      clientY: coordY - window.scrollY,
+    };
+
+    if (!this.draggingElement) {
+      return;
+    }
```

Comment:
Possibly a micro optimization, but it seems this early return could be moved to the top of the method.  

Then if there's no dragging element, it will execute the least amount of code (and not set the `options` variable to an object unless it's going to use it to dispatch the event)

Reply:
Yeah good call, I think it reads better this way too